datafusion/expr/src/function.rs

```diff
@@ -37,10 +38,36 @@ pub type ScalarFunctionImplementation =
 pub type ReturnTypeFunction =
     Arc<dyn Fn(&[DataType]) -> Result<Arc<DataType>> + Send + Sync>;
 
-/// Factory that returns an accumulator for the given aggregate, given
-/// its return datatype.
+/// Arguments passed to create an accumulator
+pub struct AccumulatorArgs<'a> {
+    // default arguments
+    pub data_type: &'a DataType, // the return type of the function
+    pub schema: &'a Schema,      // the schema of the input arguments
+    pub ignore_nulls: bool,      // whether to ignore nulls
+
+    // ordering arguments
+    pub sort_exprs: &'a [Expr], // the expressions of `order by`
```

Comment:
If you made this `Option<&'a [Expr]>` that would probably be easier for UDF implementors to check if ORDER BY was specified

Reply:
I think we can have an empty vec to describe no `ORDER BY` is given.